routes/api/events.js

```diff
@@ -0,0 +1,94 @@
+const mongoose = require('mongoose');
+const express = require("express");
+const router = express.Router();
+const Event = require('../../models/Event');
+const validateEventInput = require('../../validations/event');
+const validateEventUpdate = require('../../validations/event');
+
+router.get("/allEvents", (req, res) => {
+  //events are being sent up as an array
+    Event.find().then(events => {
+        res.send(events); 
+    }).catch(err => console.log(err));
+});
+
+
+router.get("/:id", (req, res) => {
+    const event = Event.findOne({_id: req.params.id}).exec();
+    event.then(function (doc) {res.send(doc)})
+})
+
+router.post("/newEvent", (req, res) => {
+    console.log(req.body)
+    const { errors, isValid } = validateEventInput(req.body);
+
+    if (!isValid) {
+        return res.status(400).json(errors);
+    }
+    
+    Event.findOne({ name: req.body.name }).then(event => {
+        if (event) {
+            errors.name = "This event name has already been taken";
+            return res.status(400).json(errors);
+        } else {
+          const newEvent = new Event({
+            name: req.body.name,
+            description: req.body.description,
+            lat: req.body.lat,
+            long: req.body.long,
+            imageUrl: req.body.imageUrl,
+            time: req.body.time
+        })
+        newEvent.save().then(newEvent => {
+            res.json(newEvent)
+        }).catch(err => console.log(err))
+        
+    }
+    
+    
+}).catch(err => console.log(err))
+});
+
+router.patch("/:id", (req, res) => {
```

Comment:
your update route doesn't seem to be working, I tried to update name and description for the existing two records and they don't work. I got {} in my console so I'm not sure what's wrong